contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -106,12 +116,16 @@ func (h *requestOrResponseHeaderMapImpl) Del(key string) {
 	// Get all header values first before removing a key, since the del operation may not take affects immediately
 	// when it's invoked in a Go thread, instead, it will post a callback to run in the envoy worker thread.
 	// Otherwise, we may get outdated values in a following Get call.
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
 	h.initHeaders()
 	delete(h.headers, key)
 	cAPI.HttpRemoveHeader(unsafe.Pointer(h.request.req), &key)
 }
 
 func (h *requestOrResponseHeaderMapImpl) Range(f func(key, value string) bool) {
+	h.mutex.Lock()
```

Comment:
oh, we should be careful here, since people may use `header.Del` in the `f` function, that will leads to dead lock.

Reply:
yeah, some comments are added to avoid dead lock temporarily. PTAL, thanks.

Reply:
What about running the `f` over a copy of headers? It would be a disaster to be deadlocked in prod.

Reply:
yep, that's a solution, but it may introduce much more performance cost. maybe we could provide two methods, one for better performance and one for safety.
And we'd better benchmark them before we choosing, @StarryVae do you want to have a try?

Reply:
sorry for late, i will fix it soon.